graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/InputObjectMapper.kt

```diff
@@ -104,4 +88,24 @@ object InputObjectMapper {
 
         return instance
     }
+
+    private fun convertList(input: List<*>, nestedTarget: KClass<*>): List<*> {
+        return input.filterNotNull().map { listItem ->
+            if (nestedTarget.java.isAssignableFrom(listItem::class.java)) {
+                listItem
+            } else if (nestedTarget.java.isEnum) {
```

Comment:
We need to ask if the target class is an enum, and if such we need to fetch the enum value. If we don't do this the resulting collection/list will have strings instead of the enums.